python/cudf/cudf/_libxx/join.pyx

```diff
@@ -0,0 +1,139 @@
+from cudf._libxx.lib cimport *
+from cudf._libxx.table cimport Table
+from cudf._libxx.includes.join cimport *
+from libcpp.vector cimport vector
+from libcpp.pair cimport pair
+
+from collections import OrderedDict
+
+cpdef join(lhs, rhs, left_on, right_on, how, method, left_index=False, right_index=False):
+    """
+    Call libcudf++ join for full outer, inner and left joins.
+    Returns a list of tuples [(column, valid, name), ...]
+    """
+
+    cdef vector[int] left_on_ind
+    cdef vector[int] right_on_ind
```

Comment:
We use `push_back` on these later where it would be good to reserve the space up front if we know the size so it doesn't have to keep resizing the vectors

Reply:
Agreed, these should `reserve` up front now, one element for each merge column and a possible extra for the index.